### PR TITLE
Fix rounding error when displaying fee percentages

### LINKED ISSUE #2059: Base fee for Australian accounts incorrect on Overview screen
With version 2.5.0, the account details card on the **Payments > Overview** screen displays an incorrect % value in the **Base Fee** for an Australian account.

The fee displayed is 1.8%, when the correct base fee is 1.75% (presumably this a rounding error).

wcpay-overview-australian-fees

### PR DESCRIPTION
**Pull request by aprea**
Fixes #2059 

#### Changes proposed in this Pull Request

Fixes an issue where fees were rounded to a single decimal place. This resulted in the fee for Australian merchants displaying as `1.8%` rather than `1.75%`.

This PR introduces the following changes:
- Display fees with the minimum number of decimal places possible (e.g. `3%` rather than `3.0%` or `3.00%`).
- Display fees with a maximum of 3 decimal places (e.g. `2.175%`).

|Before|After|
|-|-|
|fee-before|fee-after|

I have also introduced an unrelated (but necessary) change to the `lint-staged` config.

I have moved the config from `package.json` to `lint-staged.config.js`.

Prior to this change, I observed the following error when attempting to commit my changes:

```
✔ Preparing...
⚠ Running tasks...
  ✔ Running tasks for *.{js,jsx,ts,tsx}
  ❯ Running tasks for *.{ts,tsx}
    ✖ tsc --noEmit --jsx react [FAILED]
  ↓ No staged files match *.{scss,css} [SKIPPED]
  ↓ No staged files match *.php [SKIPPED]
  ↓ No staged files match composer.json [SKIPPED]
↓ Skipped because of errors from tasks. [SKIPPED]
✔ Reverting to original state because of errors...
✔ Cleaning up...

✖ tsc --noEmit --jsx react:
client/utils/fees/index.test.ts(4,27): error TS2307: Cannot find module 'utils/fees' or its corresponding type declarations.
husky - pre-commit hook exited with code 1 (error)
```

This is due to a known incompatibility between `lint-staged` and TypeScript (`tsc`), more information about this here - https://github.com/microsoft/TypeScript/issues/27379

I have implemented this solution - https://github.com/okonet/lint-staged/issues/825#issuecomment-620018284

#### Testing instructions

**Scenario 1**
* Navigate to the WCPay overview page `/wp-admin/admin.php?page=wc-admin&path=%2Fpayments%2Foverview`
* Observe the fee should display accurately in the "Account details" section.

**Scenario 2**
* Navigate to the **Transactions > Payment details** page.
* Observe the fee should display accurately in the "Timeline" section.

-------------------

- [x] Added changelog entry (or does not apply)
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

### CODE REVIEW

**Comment by aprea** [comment]
Hmm, odd, the fee is showing as `2.9%` in the payment timeline in my local environment but showing as `1.75%` on the WCPay overview page.
Screen Shot 2021-06-15 at 3 50 19 pm

Maybe I'm missing some config somewhere?

File: client/payment-details/timeline/map-events.js [hunk 1]
```diff
@@ -194,9 +195,9 @@ const composeFeeString = ( event ) => {
 
 	return sprintf(
 		/* translators: %1$s is the total fee amount, %2$f%% is the fee percentage, and %3$s is the fixed fee amount. */
```
Comment (shendy-a8c) [inline client/payment-details/timeline/map-events.js:197]: Interesting that this comment had been inconsistent with the code. Notice `%2$f%%` and `%2$.1f%%` (before your change). Your change aligns with the comment =)

**Review by shendy-a8c — APPROVED** [review]
LGTM. Just need to add changelog to `readme.txt`. Also, don't forget to sync with base branch before merging.

**Comment by vbelolapotkov** [comment]
@shendy-a8c does it need final review? If not, could you please set ready label?

**Comment by shendy-a8c** [comment]
It's ready to merge. I have updated the label. Sorry for the mix-up @aprea you'll need to sync with base branch again.

**Pull request merged**